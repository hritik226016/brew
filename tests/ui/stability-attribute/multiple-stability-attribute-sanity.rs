//! Checks that multiple stability attributes are used correctly together

#![feature(staged_api)]

#![stable(feature = "stable_test_feature", since = "1.0.0")]

#[unstable(feature = "a", issue = "none", reason = "reason 1")]
#[unstable(feature = "b", issue = "none", reason = "reason 2")] //~ ERROR multiple reasons provided for unstability
fn f1() { }

#[unstable(feature = "a", issue = "none", reason = "reason 1")]
#[unstable(feature = "b", issue = "none", reason = "reason 2")] //~ ERROR multiple reasons provided for unstability
#[unstable(feature = "c", issue = "none", reason = "reason 3")] //~ ERROR multiple reasons provided for unstability
fn f2() { }

#[unstable(feature = "a", issue = "none")] //~ ERROR `soft` must be present on either none or all of an item's `unstable` attributes
#[unstable(feature = "b", issue = "none", soft)]
fn f3() { }

#[unstable(feature = "a", issue = "none", soft)] //~ ERROR `soft` must be present on either none or all of an item's `unstable` attributes
#[unstable(feature = "b", issue = "none")]
fn f4() { }

fn main() { }
