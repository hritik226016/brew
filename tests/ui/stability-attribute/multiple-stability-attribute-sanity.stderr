error: multiple reasons provided for unstability
  --> $DIR/multiple-stability-attribute-sanity.rs:8:1
   |
LL | #[unstable(feature = "b", issue = "none", reason = "reason 2")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: multiple reasons provided for unstability
  --> $DIR/multiple-stability-attribute-sanity.rs:12:1
   |
LL | #[unstable(feature = "b", issue = "none", reason = "reason 2")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: multiple reasons provided for unstability
  --> $DIR/multiple-stability-attribute-sanity.rs:13:1
   |
LL | #[unstable(feature = "c", issue = "none", reason = "reason 3")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `soft` must be present on either none or all of an item's `unstable` attributes
  --> $DIR/multiple-stability-attribute-sanity.rs:16:1
   |
LL | #[unstable(feature = "a", issue = "none")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
LL | #[unstable(feature = "b", issue = "none", soft)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `soft` must be present on either none or all of an item's `unstable` attributes
  --> $DIR/multiple-stability-attribute-sanity.rs:20:1
   |
LL | #[unstable(feature = "a", issue = "none", soft)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
LL | #[unstable(feature = "b", issue = "none")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 5 previous errors

